frontend/src/pages/ExperimentDetails.test.tsx

```diff
@@ -0,0 +1,453 @@
+    await TestUtils.flushPromises();
+    expect(updateToolbarSpy).toHaveBeenLastCalledWith(expect.objectContaining({
+      pageTitle: 'test exp ID',
+      pageTitleTooltip: 'test exp ID'
+    }));
+    tree.unmount();
+  });
+
+  it('uses the experiment name as the page title', async () => {
+    const experiment = newMockExperiment();
+    experiment.name = 'A Test Experiment';
+
+    getExperimentSpy.mockImplementationOnce(() => experiment);
+
+    const tree = shallow(<ExperimentDetails {...generateProps()} />);
+    await TestUtils.flushPromises();
+    expect(updateToolbarSpy).toHaveBeenLastCalledWith(expect.objectContaining({
+      pageTitle: 'A Test Experiment',
+      pageTitleTooltip: 'A Test Experiment'
+    }));
+    tree.unmount();
+  });
+
+  it('uses an empty string if the experiment has no description', async () => {
+    const experiment = newMockExperiment();
+    experiment.description = undefined;
+
+    getExperimentSpy.mockImplementationOnce(() => experiment);
+
+    const tree = shallow(<ExperimentDetails {...generateProps()} />);
+    await TestUtils.flushPromises();
+    expect(tree).toMatchSnapshot();
+    tree.unmount();
+  });
+
+  it('removes all description text after second newline and replaces with an ellipsis', async () => {
+    const experiment = newMockExperiment();
+    experiment.description = 'Line 1\nLine 2\nLine 3\nLine 4';
+
+    getExperimentSpy.mockImplementationOnce(() => experiment);
+
+    const tree = shallow(<ExperimentDetails {...generateProps()} />);
+    await TestUtils.flushPromises();
+    expect(tree).toMatchSnapshot();
+    tree.unmount();
+  });
+
+  it('opens the expanded description modal when the expand button is clicked', async () => {
+    const tree = TestUtils.mountWithRouter(<ExperimentDetails {...generateProps() as any} />);
+    await TestUtils.flushPromises();
+
+    tree.update();
+
+    tree.find('#expandExperimentDescriptionBtn').at(0).simulate('click');
+    await TestUtils.flushPromises();
+    expect(updateDialogSpy).toHaveBeenCalledWith({
+      content: MOCK_EXPERIMENT.description,
+      title: 'Experiment description',
+    });
+    tree.unmount();
+  });
+
+  it('calls getExperiment with the experiment ID in props', async () => {
+    const props = generateProps();
+    props.match = { params: { eid: 'test exp ID' } } as any;
+    const tree = shallow(<ExperimentDetails {...props} />);
+    await TestUtils.flushPromises();
+    expect(getExperimentSpy).toHaveBeenCalledTimes(1);
+    expect(getExperimentSpy).toHaveBeenCalledWith('test exp ID');
+    tree.unmount();
+  });
+
+  it('shows an error banner if fetching the experiment fails', async () => {
+    TestUtils.makeErrorResponseOnce(getExperimentSpy, 'test error');
+
+    const tree = shallow(<ExperimentDetails {...generateProps()} />);
+    await TestUtils.flushPromises();
+
+    expect(updateBannerSpy).toHaveBeenLastCalledWith(expect.objectContaining({
```

Comment:
Also expect it to have been called once to clear it on init?

Reply:
I think this is covered by your other comment above?

Reply:
It is.